lib/vmpooler/vsphere_helper.rb

```diff
@@ -196,39 +168,96 @@ def find_folder(foldername)
       base
     end
 
-    def find_least_used_host(cluster)
+    def get_host_utilization(host, model=nil, limit=90)
+      # Returns an array containing cumulative CPU and memory utilization of a host, and its object reference
+      # Params:
+      # +model+:: CPU arch version to match on
+      # +limit+:: Hard limit for CPU or memory utilization beyond which a host is excluded for deployments
+      if model
+        return nil unless host_has_cpu_model? host, model
+      end
+      return nil if host.runtime.inMaintenanceMode
+      return nil unless host.overallStatus == 'green'
+
+      cpu_utilization = cpu_utilization_for host
+      memory_utilization = memory_utilization_for host
+
+      return nil if cpu_utilization > limit
+      return nil if memory_utilization > limit
+
+      [ cpu_utilization + memory_utilization, host ]
+    end
+
+    def host_has_cpu_model?(host, model)
+       get_host_cpu_arch_version(host) == model
+    end
+
+    def get_host_cpu_arch_version(host)
+      cpu_model = host.hardware.cpuPkg[0].description
+      cpu_model_parts = cpu_model.split()
+      arch_version = cpu_model_parts[4]
+      arch_version
+    end
+
+    def cpu_utilization_for(host)
+      cpu_usage = host.summary.quickStats.overallCpuUsage
+      cpu_size = host.summary.hardware.cpuMhz * host.summary.hardware.numCpuCores
+      (cpu_usage.to_f / cpu_size.to_f) * 100
+    end
+
+    def memory_utilization_for(host)
+      memory_usage = host.summary.quickStats.overallMemoryUsage
+      memory_size = host.summary.hardware.memorySize / 1024 / 1024
+      (memory_usage.to_f / memory_size.to_f) * 100
+    end
+
+    def vsphere_connection_alive?(connection)
```

Comment:
> It verifies a connection is alive reconnects when it is not.

Ahh, I couldn't even tell that when looking at the definition itself (I had no idea what that `initialize` was going to do).

I think it would be good to reflect the behavior in the name, so something like `ensure_connected(connection)` or `reconnect_if_dead(connection)` or something like that.